src/4D_api.cpp

```diff
@@ -189,6 +189,11 @@ double proj_roundtrip(PJ *P, PJ_DIRECTION direction, int n, PJ_COORD *coord) {
     /* finally, we take the last half-step */
     t = proj_trans(P, opposite_direction(direction), t);
 
+    /* if we start with NaN, we expect NaN as output */
+    if (isnan(org.v[0]) && isnan(t.v[0])) {
```

Comment:
Please use std::isnan() here and in other places (and make sure there's a #include <cmath>)

Reply:
Ok this should be done now